src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -2051,16 +2054,16 @@ static QVariant fcnIsSelected( const QVariantList &values, const QgsExpressionCo
   if ( values.isEmpty() )
   {
     feature = context->feature();
-    layer = QgsExpressionUtils::getVectorLayer( context->variable( QStringLiteral( "layer" ) ), parent );
+    layer = QgsExpressionUtils::getVectorLayer( context->variable( QStringLiteral( "layer" ) ), context, parent );
```

Comment:
Should this be run via `runMapLayerFunctionThreadSafe`? (see https://github.com/qgis/QGIS/pull/51176/files?w=1#diff-1edfa3cc15da06ba954e37e8662a5bf8ad502e5e0c3b10b055e111bbaa83d18eL2077)

Reply:
Yes -- I've only updated a handful of the functions here. There's still around 20 or so which aren't updated.